### ISSUE TITLE
Drag is being terribly scaled!

### ISSUE THREAD

**Issue by Lisias** [net-lisias-ksp/TweakScale#155]
Our fellow Kerbonaut [Lordakos](https://github.com/Lordakos) raised a very important [issue](https://github.com/net-lisias-ksp/TweakScale/issues/92#issuecomment-754134517):

> Please correct me if im wrong but I've checked the terminal velocities and the 2x bigger one has around 10X higher terminal velocity simulated at the same speed and altitude than the smaller one. that cannot be wright. Here me out: F=m.a of course and Fdrag is coeff.vˇ2.A.density.0,5 if the scale is half the A is 0,25 all others beeing equal. now the the mass at half size is 1/8 the original because 0,5^3 so the deacceleration of the thing at at half size should be 0,25F=1/8m.a the new a is 2x the full sized a. this is now what I get in game.
Same thing can be seen with terminal velocity. Fdrag=Fgravity at full size coeff.vˇ2.A.density.0,5=m.9,81
at half the size coeff.vˇ2.0,25A.density.0,5=1/8m.9,81 therefore vterminalhalf=1/sqr(2).vterminalfull
but instead I get vterminalhalf=1/10.vterminalfull.
>
> This is way too much of a difference and it is very noticable. Either Iam really bad at math(would not be surprising) or tweakscale doesnt scale drag proparly.
>
> Even if I am wring the mass issue still insists.

My initial [response](https://github.com/net-lisias-ksp/TweakScale/issues/92#issuecomment-754338034) was not unreasonable, but [I was proved wrong](https://github.com/net-lisias-ksp/TweakScale/issues/92#issuecomment-754348690). **The drag is being scaled terribly wrong**!

I made this stunt:
![screenshot32](https://user-images.githubusercontent.com/64334/103599953-144d0000-4ee5-11eb-9195-d16126957663.png)

It have a default X200-16 tank on one side, and a scaled down to 2.5 S4-128 on the other side. Both empty.

The difference on weight is negligible to our purposes:
![Screen Shot 2021-01-04 at 23 28 39](https://user-images.githubusercontent.com/64334/103600095-67bf4e00-4ee5-11eb-912d-faf850dfe489.png)

And put it to fly:
![screenshot33](https://user-images.githubusercontent.com/64334/103599985-2038c200-4ee5-11eb-9b4d-b753f964558d.png)

And... yes. Both tanks should have approximately the same amount the drag, but the scaled down one is incredibly less draggy. What also implies that scaled up ones are terribly more draggy than they should.

Geez.... This is going to bite. o.O Badly.

This need to be tackled down for sure, it's spoiling terribly the game and giving a unfair hand on challenges (by helping up scaled down parts, and improperly taxing scaled up ones).

On a note to myself, I probably should cook a way to allow the user to use the old drag scaling as an option....

**Comment by Lisias** [comment]
New test, this time scaling up a smaller part.

I took a Rockomax X200-16 (empty), and a scaled up FL-T200 to match diameters. Both parts scaled up to 1ton each (empty)

![screenshot49](https://user-images.githubusercontent.com/64334/103956802-0804b500-5128-11eb-8a0c-d24474dccf34.png)

And when firing up the thing:

![screenshot51](https://user-images.githubusercontent.com/64334/103956919-4a2df680-5128-11eb-9f14-d1fe2f839d8a.png)

I noticed that the scaled up part is **way** more draggy than the not scaled part.

**Comment by Lisias** [comment]
For benchmarking, I redid the last test using unscaled parts:

![screenshot52](https://user-images.githubusercontent.com/64334/103957093-b577c880-5128-11eb-955d-a900fee15502.png)

And fire it up to see what I get:

![screenshot54](https://user-images.githubusercontent.com/64334/103957367-59fa0a80-5129-11eb-9363-083936b104c0.png)

And yep, too much frag on scaled up parts.

Just for the sake of completude, I scaled up again the T200:

![screenshot55](https://user-images.githubusercontent.com/64334/103957433-77c76f80-5129-11eb-8bca-aa566d682fae.png)

And then reset and scaled down the X200:

![screenshot56](https://user-images.githubusercontent.com/64334/103957543-b2310c80-5129-11eb-81c3-aa28f47212f3.png)

The inconsistency is deterministic, it's not something happening randomly - its something wrong on the drag scaling algorithm.

**Comment by Lisias** [comment]
Unsurprisingly, I reproduced the problem downto KSP 1.4.4 . This eliminates any possibility that this could be something related to KSP itself.

And the problem is way problematic than I thought.

**Comment by Lisias** [comment]
Since the behaviour is consistent on KSP 1.11.0 and 1.4.4., I will pursue the problem on 1.4.4 as this version is **way** faster on my rig (including loading).

I reproduced the tests above using TweakScale 2.4.3.21:

![screenshot55](https://user-images.githubusercontent.com/64334/104007622-03231e00-5187-11eb-9222-a138923a2914.png)

![screenshot56](https://user-images.githubusercontent.com/64334/104007645-0918ff00-5187-11eb-87c2-af7b66d639ba.png)

![screenshot58](https://user-images.githubusercontent.com/64334/104007671-146c2a80-5187-11eb-94c5-092c7ca196b6.png)

The drag scaling **is all right** on 2.4.3.21 . It's something I introduced on TS 2.4.4. (sigh(.

**Comment by Lisias** [comment]
**HOLY CRAP**!! I KNEW THIS COULD HAPPEN, BUT FORGOT ABOUT!!!!

https://github.com/net-lisias-ksp/TweakScale/blob/86899cf35d86f15c73a680f2873e3a147350a6a2/Source/Scale/PartDB.cs#L138

https://github.com/net-lisias-ksp/TweakScale/blob/86899cf35d86f15c73a680f2873e3a147350a6a2/Source/Scale/PartDB.cs#L528

I introduced this thing when I implemented scaling Cost and Mass on Variants!!!!

https://github.com/net-lisias-ksp/TweakScale/commit/7596af4e5f300ea9e3bf74a07fcc12f4a4d9443c

**Comment by Lisias** [comment]
I remembered why I did that.

I was facing some problems on changing variants after scaling, and wondered if drags would not be affected by changing variants as it was happening with some other parameters. I intended to do some tests like these ones at that time, but something else must had happened and I was diverted from the task - and forgot about. (sigh).

Well, I was wrong on the variants affecting negatively the scaled drags.

This regression was fixed on commit: https://github.com/net-lisias-ksp/TweakScale/commit/63fa47c67842233ad8fc9aab0bed1258320b30bf

![screenshot57](https://user-images.githubusercontent.com/64334/104010491-539c7a80-518b-11eb-98a8-1f5630c72f15.png)
